// @riccox/colorify/dist/index.mjs
// src/palette.ts
import chroma from "chroma-js";
var calculateColorPalette = (hue, sat = 0.75, numOfShades = 12) => {
  return chroma.scale(
    [
      0.97,
      0.94,
      0.86,
      0.77,
      0.66,
      0.5,
      0.45,
      0.39,
      0.33,
      0.27,
      0.21,
      0.11,
      0.03
    ].map((lum) => chroma.hsl(hue, sat, lum))
  ).colors(numOfShades, "hex");
};
var singleColorPalette = (color, numOfShades = 12) => {
  const [hue, sat] = chroma(color).hsl();
  return calculateColorPalette(hue, sat, numOfShades);
};
var generateRadixColorPalette = (color, colorSchema = "light") => {
  const [hue, sat] = chroma(color).hsl();
  const lumScaleLight = [
    0.978,
    0.938,
    0.902,
    0.868,
    0.832,
    0.804,
    0.747,
    0.659,
    0.541,
    0.453,
    0.27,
    0.086
  ];
  const lumScaleDark = [
    0.07,
    0.11,
    0.136,
    0.158,
    0.179,
    0.205,
    0.243,
    0.313,
    0.439,
    0.52,
    0.61,
    0.93
  ];
  const lumScaleWhite = [
    1,
    0.975,
    0.94,
    0.905,
    0.875,
    0.845,
    0.815,
    0.77,
    0.18,
    0.13,
    0.07,
    0
  ];
  const lumScaleBlack = [
    0,
    0.035,
    0.075,
    0.1,
    0.12,
    0.155,
    0.19,
    0.225,
    0.91,
    0.94,
    0.975,
    1
  ];
  if (hue > 0 && sat > 0) {
    return (colorSchema === "dark" ? lumScaleDark : lumScaleLight).map(
      (lum) => chroma.hsl(hue, sat, lum).hex()
    );
  }
  return (colorSchema === "dark" ? lumScaleBlack : lumScaleWhite).map(
    (lum) => chroma.hsl(0, 0, lum).hex()
  );
};
var getBlackWhiteColorPalette = (numOfShades = 12) => {
  return chroma.scale().colors(numOfShades, "hex");
};

// src/foreground.ts
import chroma2 from "chroma-js";
var foregroundColor = (bgColor) => {
  const [r, g, b] = chroma2(bgColor).rgb();
  const sum = Math.round((r * 299 + g * 587 + b * 114) / 1e3);
  return sum > 128 ? "#000000" : "#FFFFFF";
};
var foregroundColorViaLum = (bgColor) => {
  const luma = chroma2(bgColor).luminance();
  const alpha = chroma2(bgColor).alpha();
  let targetLum = luma > 0.5 ? luma - 0.5 : luma + 0.5;
  if (targetLum > 0.5 && alpha < 0.5)
    targetLum = 0.4;
  return chroma2(bgColor).luminance(targetLum).hex();
};

// src/complementary.ts
var complement = (color) => {
  const colorPart = color.slice(1);
  const ind = parseInt(colorPart, 16);
  let iter = ((1 << 4 * colorPart.length) - 1 - ind).toString(16);
  while (iter.length < colorPart.length) {
    iter = "0" + iter;
  }
  return `#${iter}`;
};
export {
  calculateColorPalette,
  complement,
  foregroundColor,
  foregroundColorViaLum,
  generateRadixColorPalette,
  getBlackWhiteColorPalette,
  singleColorPalette
};
